### ISSUE TITLE
Non-integer numbers cause incorrect OpenApi YAML 

### ISSUE THREAD

**Issue by gbogard** [softwaremill/tapir#207]
When a json Input has non-integer fields (double or float) and one provides an example for the input, those fields in the example get serialized as `!!int '<the value>'` when interpreting the endpoint as OpenAPI YAML.

This is causing Swagger to refuse the documentation. (See [screenshot](https://i.imgur.com/31Ic8LQ.png))

## Reproduction

```scala
import tapir._
import tapir.docs.openapi._
import tapir.swagger.http4s.SwaggerHttp4s
import tapir.openapi.circe.yaml._
import tapir.openapi.Info
import tapir.json.circe._
import cats.effect._
import io.circe.generic.auto._

object Documentation {

  case class Request(height: Float, Width: Float)

  val myEndPoint = endpoint
    .post
    .in(jsonBody[Request].example(Request(45, 52)))
    .out(stringBody)

  val info = Info(
    "My API",
    ""
  )

  val yaml                                    = myEndPoint.toOpenAPI(info).toYaml
  def routes()(implicit cs: ContextShift[IO]) = new SwaggerHttp4s(yaml).routes[IO]
}
```

The resulting YAML :
```
openapi: 3.0.1
info:
  title: My API
  version: ''
paths:
  /:
    post:
      operationId: postRoot
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Request'
            example:
              height: !!int '45.0'
              Width: !!int '52.0'
        required: true
      responses:
        '200':
          description: ''
          content:
            text/plain:
              schema:
                type: string
components:
  schemas:
    Request:
      required:
      - height
      - Width
      type: object
      properties:
        height:
          type: number
        Width:
          type: number
```

I believe the `!!int '45.0'` is causing the issues with Swagger. [This issue](https://github.com/circe/circe-yaml/issues/68) on the Circe-yaml project must be closely related. Any idea how I can provide examples for such types ?

**Comment by adamw** [comment]
@gbogard the temporary work-around is to provide non-whole example values, e.g. `Request(45.9, 52.1)`

**Comment by gbogard** [comment]
We'll do that for now, thanks !

**Comment by guersam** [comment]
My current workaround is using regex:

```scala
def fixDecimalExamples(s: String): String =
  s.replaceAll("!!int ('.*\\..*')", "$1")

fixDecimalExamples(myEndpoint.toOpenAPI(info).toYaml)
```

**Comment by kell18** [comment]
@guersam Thanks for the snippet! But you're treating floats as strings then. It'll respect the types:
`def swagger_fixIntToFloat(s: String): String = s.replaceAll("!!int ('\\d*\\.\\d*')", "!!float $1")`

**Comment by adamw** [comment]
Isn't this issue gone with circe-yaml 0.12.0?

**Comment by adamw** [comment]
yes, jsut tested, should be fixed, (latest is 0.12.4)
